cmd/kind/get/kubeconfig/kubeconfig.go

```diff
@@ -26,7 +26,7 @@ import (
 	"sigs.k8s.io/kind/pkg/errors"
 
 	"sigs.k8s.io/kind/pkg/cluster"
-	clusternodes "sigs.k8s.io/kind/pkg/cluster/nodes"
+	clusternodes "sigs.k8s.io/kind/pkg/internal/cluster/nodes"
```

Comment:
yeah...this will break kinder, so we have to "third_party" it.
cc @fabriziopandini

Reply:
see the later commits

Reply:
we still support:
- provisioning a cluster 
- getting some node details and execing against it

however the API itself breaks, and some of the utilities for interacting with nodes may become public or private over time (to reflect how stable they are)

each of those are much smaller though, the bulk of the complicated code here (provisioning "machines", supporting the node interface to exec etc.) will stay usable via public APIs, but I think those need some more reshaping (cluster.Context is weird now and needs a slight rethink, but should provide the same functionality ish)

Reply:
again, this PR will _not_ be the final state for that as it's just way too big of a change, `cmd/` still imports the old docker package for example, I will ~eliminate that but ensure that the same level of power is available through a more deliberate API.

Reply:
ok, i saw the later commits.
understood.

Reply:
I should probably squash this PR, the commits are wildly non-meaningful, I spent a bit of time refactoring the history but didn't come up with a clean version with small commits